packages/fuel-indexer-graphql/src/dynamic.rs

```diff
@@ -0,0 +1,457 @@
+use std::collections::{HashMap, HashSet};
+
+use async_graphql::{
+    dynamic::{
+        Enum, Field, FieldFuture, FieldValue, InputObject, InputValue, Object,
+        ResolverContext, Scalar, Schema as DynamicSchema,
+        SchemaBuilder as DynamicSchemaBuilder, TypeRef,
+    },
+    Request,
+};
+use async_graphql_parser::types::{BaseType, Type};
+use fuel_indexer_database::{queries, IndexerConnectionPool};
+use fuel_indexer_schema::db::tables::Schema;
+use lazy_static::lazy_static;
+use serde_json::Value;
+
+use crate::graphql::{GraphqlError, GraphqlQueryBuilder};
+
+lazy_static! {
+    static ref SCALAR_TYPES: HashSet<&'static str> = HashSet::from([
+        "ID",
+        "Address",
+        "AssetId",
+        "Bytes4",
+        "Bytes8",
+        "Bytes32",
+        "Bytes64",
+        "Int4",
+        "Int8",
+        "Int16",
+        "UInt4",
+        "UInt8",
+        "UInt16",
+        "Timestamp",
+        "Color",
+        "ContractId",
+        "Salt",
+        "Json",
+        "MessageId",
+        "Charfield",
+        "Identity",
+        "Boolean",
+        "Blob",
+    ]);
+    static ref NUMERIC_SCALAR_TYPES: HashSet<&'static str> = HashSet::from([
+        "Int4",
+        "Int8",
+        "Int16",
+        "UInt4",
+        "UInt8",
+        "UInt16",
+        "Timestamp",
+    ]);
+    static ref STRING_SCALAR_TYPES: HashSet<&'static str> = HashSet::from([
+        "ID",
+        "Address",
+        "AssetId",
+        "Bytes4",
+        "Bytes8",
+        "Bytes32",
+        "Bytes64",
+        "Color",
+        "ContractId",
+        "Salt",
+        "Json",
+        "MessageId",
+        "Charfield",
+        "Identity",
+        "Blob",
+    ]);
+    static ref SORTABLE_SCALAR_TYPES: HashSet<&'static str> = HashSet::from([
+        "ID",
+        "Address",
+        "AssetId",
+        "Int4",
+        "Int8",
+        "Int16",
+        "UInt4",
+        "UInt8",
+        "UInt16",
+        "Timestamp",
+        "Color",
+        "ContractId",
+        "Salt",
+        "MessageId",
+        "Charfield",
+        "Identity",
+    ]);
+    static ref IGNORED_ENTITY_TYPES: HashSet<&'static str> =
+        HashSet::from(["IndexMetadataEntity", "QueryRoot"]);
+    static ref IGNORED_ENTITY_FIELD_TYPES: HashSet<&'static str> =
+        HashSet::from(["object"]);
+}
+
+/// Execute user query and return results.
+pub async fn execute_query(
+    dynamic_request: Request,
+    dynamic_schema: DynamicSchema,
+    user_query: String,
+    pool: IndexerConnectionPool,
+    schema: Schema,
+) -> Result<Value, GraphqlError> {
```

Comment:
```suggestion
) -> GraphQLResult<Value, GraphqlError> {
```
(Following our pattern)